enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,430 @@
+# TYPE cluster_installation_ignition counter
+cluster_installation_ignition{platform="aws",result="success",version="4.2",os="linux"} 1
+```
+
+Possible `result` values are `success` and are `error`, otherwise all possible label values are the same as `cluster_installation_create`. The sample value would always be one in order to aggregate to a total count of runs of this command.
+
+##### cluster_installation_destroy
+
+When a user destroys a cluster a single sample of this metric would be pushed:
+
+```
+# HELP cluster_installation_destroy represents a single run of destroy cluster by the OpenShift installer.
+# TYPE cluster_installation_destroy histogram
+cluster_installation_destroy_bucket{platform="aws",result="success",version="4.2",os="linux",le="5"} 0
+cluster_installation_destroy_bucket{platform="aws",result="success",version="4.2",os="linux",le="10"} 1
+cluster_installation_destroy_bucket{platform="aws",result="success",version="4.2",os="linux",le="15"} 1
+cluster_installation_destroy_bucket{platform="aws",result="success",version="4.2",os="linux",le="20"} 1
+cluster_installation_destroy_bucket{platform="aws",result="success",version="4.2",os="linux",le="25"} 1
+cluster_installation_destroy_bucket{platform="aws",result="success",version="4.2",os="linux",le="30"} 1
+cluster_installation_destroy_bucket{platform="aws",result="success",version="4.2",os="linux",le="+Inf"} 1
+cluster_installation_destroy_sum{platform="aws",result="success",version="4.2",os="linux"} 10
+cluster_installation_destroy_count{platform="aws",result="success",version="4.2",os="linux"} 1
+```
+The label values are the same as `openshift_installation_create` except `le` has values between 5 and 30 minutes and `result` is `success` or `error`.
+
+##### cluster_installation_gather
+We will count invocations and outcomes of `gather bootstrap`:
+
+```
+# HELP cluster_installation_gather represents a single run of gather bootstrap by the OpenShift installer.
+# TYPE cluster_installation_gather counter
+cluster_installation_gather{platform="aws",result="success",version="4.2",os="linux"} 1
+```
+Potential label values would be the same as `cluster_installation_manifests` except `result` could return an extra value for `SSHError` which shows that users tried to connect to gather, but were unauthorized.
+
+#### Duration
+
+The `cluster_installation_create` metric above captures the overall duration of an installation. The metrics in this category represent the duration of  each stage of installation in the execution of the `create` or `wait-for` commands. The proposed metrics are: 
+
+* `cluster_installation_duration_infra`
+* `cluster_installation_duration_api`
+* `cluster_installation_duration_bootstrap`
+* `cluster_installation_duration_operators`
+
+Each of these metrics (except infrastructure) would have five labels:
+  * `command` - either `create` or `wait-for`
```

Comment:
The semantics of this for `wait-for` are not clear to me.  Is `cluster_installation_duration_api{command="wait-for"}` supposed to be "since calling [`waitForBootstrapComplete`][1]" or "since something we expect to have been the previous step in a user-provided-infrastructure flow, e.g. `create ignition-configs`"?

[1]: https://github.com/openshift/installer/blob/b518ee89d4ee64f95b3b9401b25439ce65480189/cmd/openshift-install/waitfor.go#L47

Reply:
It is supposed to be since `openshift install wait-for <target>` was called.